src/commcare_cloud/ansible/roles/elasticsearch/tasks/misc_v2.yml

```
@@ -40,3 +40,8 @@
   become: yes
   shell: "{{ elasticsearch_home }}/bin/plugin install https://github.com/dimagi/elasticsearch-repository-swift/releases/download/swift-repository-plugin-{{ swift_plugin_version }}/repository-swift-3.0.0-es{{ swift_plugin_version }}.zip"
   when: (backup_es_swift) and (installed_plugins.stdout is defined) and ('repository-swift' not in installed_plugins.stdout) and elasticsearch_version == '2.4.6'
+
+- name: Install elasticsearch phonetic plugin
+  become: yes
+  shell: "{{ elasticsearch_home }}/bin/plugin install analysis-phonetic"
+  when: (backup_es_s3) and (installed_plugins.stdout is defined) and ('analysis-phonetic' not in installed_plugins.stdout) and elasticsearch_version == '2.4.6'
```

this isn't related to S3 backup so no need to check that var. It could be worth adding a var for this plugin specifically so that this can be merged but only enabled on staging.

```suggestion
  when: (installed_plugins.stdout is defined) and ('analysis-phonetic' not in installed_plugins.stdout) and elasticsearch_version == '2.4.6'
```

---

Also, the ES version check is redundant since this whole file is excluded if ES version ~= 2.4.6: https://github.com/dimagi/commcare-cloud/blob/master/src/commcare_cloud/ansible/roles/elasticsearch/tasks/main.yml#L79-L80

---

gotcha, I'll remove those checks. As far as the checking to only enable on staging would that be something like `and (deploy_env is staging)` ?

---

You can add a var here defaulted to `false`: https://github.com/dimagi/commcare-cloud/blob/master/src/commcare_cloud/ansible/roles/elasticsearch/defaults/main.yml

And then override it in https://github.com/dimagi/commcare-cloud/blob/master/environments/staging/public.yml